src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,87 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.models.Booking;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.logging.Logger;
+
+@Service
+public class BookingService implements IBookingService {
+    private static final Logger LOGGER = Logger.getLogger(BookingService.class.getName());
+    private static final String COLLECTION_PATH = "bookings";
+
+    private Firestore db;
+
+    public void setDb(Firestore db) {
+        this.db = db;
+    }
+
+    /**
+     * Retrieves a Booking entity by its id.
+     */
+    @Override
+    public Booking getBookingByID(final UUID bookingId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (bookingId == null) throw new IllegalArgumentException();
+        DocumentReference documentReference =  db.collection(COLLECTION_PATH)
+                .document(bookingId.toString());
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot document = future.get();
+        Booking booking;
+        if(document.exists()){

```

Comment:
The explicit check for `exists` is not needed, since, as highlighted in the javadoc of the `toObject` method, it will return null when it does not exist, same as you do explicitly. Javadoc for reference:
```
  /**
   * Returns the contents of the document converted to a POJO or null if the document doesn't exist.
   *
   * @param valueType The Java class to create
   * @return The contents of the document in an object of type T or null if the document doesn't
   *     exist.
   */
  @Nullable
  public <T> T toObject(@Nonnull Class<T> valueType) {

```